games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                72..=73  => dec!("1.11"),
+                74..=75  => dec!("1.12"),
+                76..=77  => dec!("1.13"),
+                78..=79  => dec!("1.14"),
+                80..=81  => dec!("1.15"),
+                82..=83  => dec!("1.16"),
+                84..=85  => dec!("1.17"),
+                86..=87  => dec!("1.18"),
+                88..=89  => dec!("1.19"),
+                90..=91  => dec!("1.20"),
+                92..=93  => dec!("1.21"),
+                94..=95  => dec!("1.22"),
+                96..=97  => dec!("1.23"),
+                98..=99  => dec!("1.24"),
+                _ => dec!(1),
+            }
+        }
+        // Generates a Uuid, turns it into a Vector, and takes the value of a cell of that vector. Returns as Decimal from 0 to 9
+        pub fn seed_10(&mut self) -> Decimal {
+            let mut digits = Vec::new();
+            let mut seed = Runtime::generate_uuid();
+            while seed > 9 {
+                digits.push(seed % 10);
+                seed = seed / 10
+            }
+            digits.push(seed);
+            digits.reverse();
+            let point = digits[6];
+            match point {
+                0 => dec!(0),
+                1 => dec!(1),
+                2 => dec!(2),
+                3 => dec!(3),
+                4 => dec!(4),
+                5 => dec!(5),
+                6 => dec!(6),
+                7 => dec!(7),
+                8 => dec!(8),
+                9 => dec!(9),
+                _ => dec!(1),
+            }
+        }
+        // For randomly choosing between 3 options. Has a .01^19 chance of failing
+        pub fn seed_3(&mut self) -> u8 {
+            let mut digits = Vec::new();
+            let mut seed = Runtime::generate_uuid();
+            while seed > 9 {
+                digits.push(seed % 10);
+                seed = seed / 10
+            }
+            digits.push(seed);
+            digits.reverse();
+            let mut counter: usize = 0;
+            let mut counter2: usize = 1;
+            loop {
+                let point = digits[counter];
+                let point2 = digits[counter2];
+                counter += 2;
+                counter2 += 2;
+                let random_24: u128 = format!("{}{}", point, point2).parse().unwrap();
+                let choice = match random_24 {
+                    0..=32 => 1,
+                    33..=65 => 2,
+                    66..=98 => 3,
+                    _ => 4,
+                };
+                if choice != 4 {
+                    return choice
+                }
+                else {
+                    continue
+                }
+            };
+        }
+        // Takes two of the same type + level Weapon/Armor/Accessory NFT, burns them, and makes a new one based off of the first. 
+        // Takes stats from first NFT. Stats increase by 20% per upgrade
+        pub fn fuse_items(&mut self, which: u8, item_bucket: Bucket) -> Bucket {
```

Comment:
I don't believe that the caller should pass the `which` argument to the method. Instead, I believe that the type of the NFT data can be inferred once you separate them so that one resource can't use multiple NFT data types.